File: C/2020/56限/bit.c
//   bit.c
//   cl  bit.c
#include <stdio.h>

int main() {
    //
    int  i, j, k;
    i = 1; j = 1; k = i & j;
    printf("%d and %d = %d \n", i, j, k);
    //
    i = 1; j = 0; k = i & j;
    printf("%d and %d = %d \n", i, j, k);
    //
    i = 10; j = 6; k = i & j;
    printf("%d and %d = %d \n", i, j, k);

    //
    i = 1; j = 1; k = i | j;
    printf("%d or %d = %d \n", i, j, k);
    //
    i = 10; j = 6; k = i | j;
    printf("%d or %d = %d \n", i, j, k);

    //
    i = 15; j = i >> 1;
    printf("%d >>1 = %d \n", i, j);
    i = -10; j = i >> 1;
    printf("%d >>1 = %d \n", i, j);


    return 0;
}

File: GameServer/2023/Period34/bit.php
<?php  // bit.php

// 論理演算
var_dump(true && false);
var_dump(true || false);
echo "<br>\n";

// ビット演算
$num1 = 0xf0;
$num2 = 0x7a;
$num3 = $num1 & $num2;
$num4 = $num1 | $num2;
var_dump($num3, $num4);
echo "<br>\n";
printf("%08s <br>\n", decbin($num1));
printf("%08s <br>\n", decbin($num2));
printf("%08s <br>\n", decbin($num3));
printf("%08s <br>\n", decbin($num4));
